File: python/stacks/dailyTemperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stak = [0]
        rv = [0] * len(temperatures)
        for i, t in enumerate(temperatures):
            while stak and temperatures[stak[-1]] < t:
                prev = stak.pop()
                rv[prev] = i - prev
            stak.append(i)
        return rv

File: java/dailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
	int[] rv = new int[temperatures.length];

	Stack<Integer> stak = new Stack<>();
	for (int i = 0, n = temperatures.length; i < n; i++)
	{
	    while ((stak.size() != 0) && (temperatures[stak.peek()] < temperatures[i]))
	    {
            int prevDay = stak.pop();
            rv[prevDay] = i - prevDay;
	    }
	    stak.add(i);
	}
	return rv;
    }
}
